### DISCUSSION TITLE
Report dimension error when i try to create collection with 'metric_type': MetricType.JACCARD

### DISCUSSION THREAD

**Discussion by Wangzhencc** [milvus-io/milvus#16541] [Q&A and General discussion]
<img width="629" alt="image" src="https://user-images.githubusercontent.com/99465983/163955822-5283e9f2-a673-49fe-9440-692152e8c163.png">

The confing I choose：

<img width="544" alt="image" src="https://user-images.githubusercontent.com/99465983/163955917-96a1cf0c-57d1-4e1a-b14d-69677007aa47.png">

When i choose MetricType as  MetricType.L2, milvus can work well.

**Comment by Wangzhencc** [comment]
milvus version: 1.1.1

**Comment by JackLCL** [comment]
@Wangzhencc Please describe your problem in English.

**Reply by Wangzhencc** [reply]
ok!

**Comment by yhmo** [comment]
As we know, the insert() api requires a 2-d array as input:
```
vectors = [[1,2,3],[4,5,6],[7,8,9]]
milvus.insert(collection_name=collection_name, records=vectors)
```

The metric_type=JACCARD, so the vectors are binary vectors. We construct binary vectors in this way:

```
def gen_binary_vectors(dim, num):
    # in binary vectors, a bit represent a dimension.
    # a value of `uint8` describe 8 dimension
    vectors = [[random.randint(0, 255) for _ in range(dim//8)] for _ in range(num)]
    data = np.array(vectors, dtype='uint8').astype("uint8")
    # convert to bytes
    return [bytes(vector) for vector in data]
```

If dim=16, each vector contains two bytes, such as [24, 58]. If you input [24, 58, 97], then you will get the error "The vector dimension must be equal to the collection dimension."
